File: 05_advanced/Dynamic.cs
using System;
public class Base
{
    public void DoIt()
    {
        Console.WriteLine("Base");
    }
}

public class Derived : Base
{
    public void DoIt()
    {
        Console.WriteLine("Derived");
    }
}


class Dynamic {
  static void Main() {
    Base b = new Base();
    Derived d = new Derived();
    Base bd = new Derived();
    
    b.DoIt();
    d.DoIt();
    bd.DoIt();
    }
}

File: 05_advanced/Dynamic.java
class Base {
  static void statMethod() {
    System.out.println("Base-statMethod");
    }

  void justMethod() {
    System.out.println("Base-justMethod");
    }
  }

class Child extends Base {
  static void statMethod() {
    System.out.println("Child-statMethod");
    }

  void justMethod() {
    System.out.println("Child-justMethod");
    }
}

public class Dynamic {
   public static void main(String [] args) {
     Base b = new Child();
     b.statMethod();
     b.justMethod();
     }
}
